src/openms/source/ANALYSIS/XLMS/OPXLSpectrumProcessingAlgorithms.cpp

```diff
@@ -350,7 +364,141 @@ namespace OpenMS
         double ppm_error = (exp_spectrum[closest_exp_peak].getMZ() - theo_mz) / theo_mz * 1e6;
         ppm_error_array.emplace_back(ppm_error);
 
+        e = closest_exp_peak + 1;  // advance experimental peak to 1-after the best match
+        ++t; // advance theoretical peak
+      }
+      else if (d < 0) // exp. peak is left of theo. peak (outside of tolerance window)
+      {
+        ++e;
+      }
+      else if (d > 0) // theo. peak is left of exp. peak (outside of tolerance window)
+      {
+        ++t;
+      }
+    }
+  }
+
+  void OPXLSpectrumProcessingAlgorithms::getSpectrumAlignmentSimple(
```

Comment:
hmm ok I see your point with the two different interfaces but there seems to be a lot of code duplication.
probably doesn't get much more readable if we would squeeze that into a templated method